services/tenant-ui/frontend/src/store/utils/fetchItem.ts

```diff
@@ -0,0 +1,37 @@
+import { useTenantApi } from '../tenantApi';
+import { Ref } from 'vue';
+
+export async function fetchItem(url: string, id: string, dict: any, error: Ref<any>, loading: Ref<boolean>, params: any = {}) {
+    /*
+    This method will fetch an item from the api and add it to a dictionary in the store, 
+    to be retrieved by the primary key value later.
+
+    TODO: fix dict typing, should be a Ref<object> but my typescript linter doesn't like that.
+    */
+
+
+    const tenantApi = useTenantApi();
+    console.log(`> fetchItem(${url})`);
+    error.value = null;
+    // loading.value = true;
+    await tenantApi
+        .getHttp(url + id, params)
+        .then((res) => {
+            dict.value[id] = res.data.item;
```

Comment:
no reason for a dict as secondary storage... what will refresh that? where else is it used? i don't think fetching an item should depend on another level of store. the component using this call should worry about it and manage it.